e2e-tests/src/checkout.integration.test.js

```diff
@@ -0,0 +1,81 @@
+import {
+  payNotice,
+  selectKeyboardInput,
+  acceptCookiePolicy,
+  verifyPaymentAndGetError,
+  generateAValidNoticeCode,
+} from './utils/helpers';
+import {
+  VALID_CARD_DATA,
+  EMAIL,
+  VALID_FISCAL_CODE,
+  INVALID_FISCAL_CODE,
+  CHECKOUT_URL,
+  CARD_NUMBER_XPAY,
+} from './utils/const';
+
+describe('Checkout payment activation tests', () => {
+  /**
+   * Test input and configuration
+   */
+  const VALID_NOTICE_CODE = generateAValidNoticeCode();
+  const PA_IRRAGGIUNGIBILE_NOTICE_CODE = String(process.env.PA_IRRAGGIUNGIBILE_NOTICE_CODE);
+
+  /**
+   * Increase default test timeout (60000ms)
+   * to support entire payment flow
+   */
+  jest.setTimeout(60000);
+  jest.retryTimes(3);
+  page.setDefaultNavigationTimeout(60000);
+  page.setDefaultTimeout(60000);
+
+  beforeAll(async () => {
+    await page.goto(CHECKOUT_URL);
+    await page.setViewport({ width: 1200, height: 907 });
+    await acceptCookiePolicy();
+  });
+
+  beforeEach(async () => {
+    await selectKeyboardInput();
+  });
+
+  it('Should correctly execute a payment (xPAY)', async () => {
```

Comment:
the choice of PSP is missing, in dev it may be necessary to select a specific PSP

Reply:
psp selection phase added